Packs/XSOAR_Engineer_Training/Integrations/XSOAREngineerTraining/XSOAREngineerTraining.py

```diff
@@ -0,0 +1,369 @@
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+
+from datetime import datetime, timedelta
+from random import randrange, choice, randint
+from copy import deepcopy
+import json
+
+
+''' SAMPLE EVENTS '''
+
+SAMPLE_EVENT = {
+    "type": "url allowed",
+    "eventID": "007",
+    "urlCategory": "MALWARE",
+    "sourceIP": "10.8.8.8",
+                "occurred": "2023-01-01T00:00:01.000Z",
+                "sourceUser": "james.bond@xsoar.local",
+                "url": "https://xsoar.pan.dev/login.zip",
+                "userAgent": "Mozilla/5.0(WindowsNT6.1;WOW64;rv:27.0)Gecko/20100101Firefox/27.0"
+}
+
+TYPES = ["url allowed", "url blocked"]
+CATEGORIES = ["PHISH", "MALWARE", "SPAM"]
+
+DATE_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
+
+''' ACTIVE DIRECTORY QUERY V2 INTEGRATION '''
+
+# Lab version of AD. Simulate the ad-get-user, ad-expire-password, and ad-set-new-password responses.
+# Fun fact about data.  68% of all data is made up, but only 8% of people know that.
```

Comment:
This is funny, but for the sake of keeping the code clean, I think it could be removed.
Both lines.